portfolyo/tools/freq.py

```diff
@@ -48,22 +119,93 @@ def up_or_down(
     1
     >>> freq.up_or_down('MS', 'MS')
     0
+    >>> freq.up_or_down('QS', 'QS-APR')
+    ValueError
+
     """
-    standard_common_ts = pd.Timestamp("2020-01-01 0:00")
-    backup_common_ts = pd.Timestamp("2020-02-03 04:05:06")
-    if common_ts is None:
-        common_ts = standard_common_ts
-
-    ts1 = common_ts + pd.tseries.frequencies.to_offset(freq_source)
-    ts2 = common_ts + pd.tseries.frequencies.to_offset(freq_target)
-    if ts1 > ts2:
-        return 1
-    elif ts1 < ts2:
-        return -1
-    if common_ts == standard_common_ts:
-        # If they are the same, try with another timestamp.
-        return up_or_down(freq_source, freq_target, backup_common_ts)
-    return 0  # only if both give the same answer.
+    # check if passed freuenices are valid
+    # assert_freq_valid(freq_source)
+    # assert_freq_valid(freq_target)
+    # Compare if the freq are the same
+    if freq_source == freq_target:
```

Comment:
This check after conversion to offset? (In case e.g. freqs are "QS" and "QS-JAN")

Reply:
done, also added case-test in tests